src/cmd/singularity/cli/actions.go

```diff
@@ -269,30 +270,7 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 	environment := os.Environ()
 
 	// Clean environment
-	for _, env := range environment {
-		e := strings.SplitN(env, "=", 2)
-		if len(e) != 2 {
-			sylog.Verbosef("can't process environment variable %s", env)
-			continue
-		}
-
-		// Transpose environment
-		if strings.HasPrefix(e[0], "SINGULARITYENV_") {
-			e[0] = strings.TrimPrefix(e[0], "SINGULARITYENV_")
-		} else if IsCleanEnv {
-			continue
-		}
-
-		if e[0] == "HOME" {
```

Comment:
How does the home env variable get set if you're removing the handling from here?